app/controllers/sellers/orders_controller.rb

```diff
@@ -1,21 +1,38 @@
 module Sellers
   class OrdersController < ApplicationController
     before_action :set_order, only: [:update]
-    def seller_orders
-      @orders = Order.joins(:order_items).where(order_items: { product_id: current_seller.products.select(:id) }).where(status: :placed).distinct
 
-      render :seller_orders
+    def index
+      @orders = Order.joins(:order_items).where(order_items: { product_id: current_seller.products.select(:id) }).where(status: :placed).distinct
     end
 
+    def show
+      @orders = Order.joins(:order_items).where(order_items: { product_id: current_seller.products.select(:id) }).distinct
+      @selected_order = @orders.find_by(id: params[:id])
+
+      if @selected_order.nil?
+        redirect_to sellers_orders_path, alert: "Order not found."
```

Comment:
Refactor your code like that `redirect_to sellers_orders_path, alert: "Order not found." if @selected_order.nil?`